### DISCUSSION TITLE
Does turbo cache packages referenced from tsconfig?

### DISCUSSION THREAD

**Discussion by Kanary159357** [vercel/turborepo#4543] [Help] [answered]
We are currently using yarn workspace and looking for some awesome monorepo tools.

The problem with yarn was that I could not cache packages each and turborepo seems to achieve it easily.
But not sure that turborepo would cache the packages refered by tsconfig.

```
{
   "references":[
        { "path": "./packages/A"} 
    ]
}
```
If I build some apps that reference Package A, does turborepo skip compile tsc for packages A if there's no change in packages A?

Anyone does have experience of this?

**Comment by anthonyshew** [comment, marked as answer]
Turborepo uses the dependencies in your `package.json` files and `turbo.json` to create your Task Graph. If there are no changes in Package A or its dependencies, Turborepo will hit cache for the task.
